profile.go

```diff
@@ -0,0 +1,142 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/grafana/pyroscope-go"
+	"github.com/hamba/logger/v2"
+	"github.com/urfave/cli/v2"
+)
+
+var allProfilingTypes = []pyroscope.ProfileType{
+	pyroscope.ProfileCPU,
+	pyroscope.ProfileInuseObjects,
+	pyroscope.ProfileAllocObjects,
+	pyroscope.ProfileInuseSpace,
+	pyroscope.ProfileAllocSpace,
+	pyroscope.ProfileGoroutines,
+	pyroscope.ProfileMutexCount,
+	pyroscope.ProfileMutexDuration,
+	pyroscope.ProfileBlockCount,
+	pyroscope.ProfileBlockDuration,
+}
+
+// Tracing flag constants declared for CLI use.
+const (
+	FlagProfilingDSN      = "profiling.dsn"
+	FlagProfileUploadRate = "profiling.upload-rate"
+	FlagProfilingTags     = "profiling.tags"
+	FlagProfilingTypes    = "profiling.types"
+)
+
+// ProfilingFlags are flags that configure profiling.
+var ProfilingFlags = Flags{
+	&cli.StringFlag{
+		Name: FlagProfilingDSN,
+		Usage: "The address an configuration of the Pyroscope server, in the format " +
+			"'http://basic:auth@server:port?token=auth-token&tenantid=tenant-id'.",
+		EnvVars: []string{"PROFILING_DSN"},
```

Comment:
Why not use `strcase.ToSNAKE` ?

Reply:
The current version doesn't at the moment, but the next major will.